### PR TITLE
Chuongv/multiple call handling

### PR DESCRIPTION
**Pull request by Chuongv**
Changes in preparation of handling multiple calls.

### CODE REVIEW

File: Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m [hunk 1]
```diff
@@ -140,13 +152,27 @@ - (BOOL)sendCallControl:(OCTToxAVCallControl)control toCall:(OCTCall *)call erro
...
-                return [self.audioEngine stopAudioFlow:error];
+
+                if ((self.audioEngine.friendNumber == friend.friendNumber) &&
+                    ([self.audioEngine isAudioRunning:nil])) {
+                    return [self.audioEngine stopAudioFlow:error];
+                }
+
+                break;
             case OCTToxAVCallControlPause:
+                [self updateCall:call withStatus:OCTCallStatusPaused];
```
Comment (dvor) [inline Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m:174]: Here you can check if you are pausing call with the same friendNumber.

File: Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m [hunk 2]
```diff
@@ -276,13 +312,15 @@ - (void)toxAV:(OCTToxAV *)toxAV receiveCallAudioEnabled:(BOOL)audio videoEnabled
 
 - (void)toxAV:(OCTToxAV *)toxAV callStateChanged:(OCTToxAVCallState)state friendNumber:(OCTToxFriendNumber)friendNumber
 {
-    OCTCall *call = [self getOrCreateCallWithFriendNumber:friendNumber];
+    OCTCall *call = [self getCurrentCallForFriendNumber:friendNumber];
 
     if ((state & OCTToxAVFriendCallStateFinished) || (state & OCTToxAVFriendCallStateError)) {
 
         [self addMessageCall:call];
```
Comment (dvor) [inline Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m]: Can you please rename `addMessageCall:` to `addMessageAndDeleteCall:`. It is confusing now.

File: Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m (hunk 1 above)
Comment (Chuongv) [inline Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m:174]: I'm not sure what you mean here? I'm already doing that here?

```
    OCTFriend *friend = call.chat.friends.firstObject;

    if (! [self.toxAV sendCallControl:control toFriendNumber:friend.friendNumber error:error]) {
        return NO;
    }
```

Comment (dvor) [inline Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m:174]: Yeah, everything is okay. My bad, sorry.

**Pull request merged**